File: nan21.dnet.module.bd/nan21.dnet.module.bd.presenter/src/main/java/net/nan21/dnet/module/bd/standards/ds/model/ClassificationItemLovDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.bd.standards.ds.model;

import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeWithCodeLov;

import net.nan21.dnet.module.bd.standards.domain.entity.ClassificationItem;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = ClassificationItem.class, jpqlWhere = " e.active = true ", sort = { @SortField(field = ClassificationItemLovDs.fCODE) })
public class ClassificationItemLovDs extends
        AbstractTypeWithCodeLov<ClassificationItem> {

    public static final String fCLASSSYSTEMID = "classSystemId";

    @DsField(join = "left", path = "classSystem.id")
    private Long classSystemId;

    public ClassificationItemLovDs() {
        super();
    }

    public ClassificationItemLovDs(ClassificationItem e) {
        super(e);
    }

    public Long getClassSystemId() {
        return this.classSystemId;
    }

    public void setClassSystemId(Long classSystemId) {
        this.classSystemId = classSystemId;
    }

}

File: nan21.dnet.module.bd/nan21.dnet.module.bd.ui.extjs/src/main/webapp/src/ds/ClassificationItemLovDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.bd.standards.ds.model.ClassificationItemLovDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"clientId", type:"int", useNull:true},
		{name:"code", type:"string"},
		{name:"name", type:"string"},
		{name:"active", type:"boolean"},
		{name:"classSystemId", type:"int", useNull:true}
	]
});

Ext.define("net.nan21.dnet.module.bd.standards.ds.model.ClassificationItemLovDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"code", type:"string"},
		{name:"name", type:"string"},
		{name:"active", type:"boolean", useNull:true},
		{name:"classSystemId", type:"int", useNull:true},
		{name:"classSystemId_From",type:"int", useNull:true},
		{name:"classSystemId_To",type:"int", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.bd.standards.ds.param.ClassificationItemLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 